### PR TITLE
Sway 1.8 rebase

### PR DESCRIPTION
**Pull request by ErikReider**
This should be it! :)

### CODE REVIEW

**Comment by WillPower3309** [comment]
Mind removing all the non English readmes and updating the main readme? I'd love to support other langues but since our readme differs from upstream I'd rather not pull in the upstream readme for other languages

**Comment by ErikReider** [comment]
> Mind removing all the non English readmes and updating the main readme? I'd love to support other langues but since our readme differs from upstream I'd rather not pull in the upstream readme for other languages

Fixed :)

**Comment by ErikReider** [comment]
@WillPower3309 seems like `wlr_matrix_projection` is now deprecated. What would you recommend using?

**Comment by ErikReider** [comment]
I'm currently not able to build this on my systems due to wlroots 0.16 not being released yet on Arch. I've just been building it with wlroots as a submodule. Is this something that we should consider doing as default?

File: sway/desktop/render.c [hunk 1]
```diff
@@ -32,6 +32,7 @@
 
 struct render_data {
 	pixman_region32_t *damage;
+	float alpha;
```
Comment (WillPower3309) [inline sway/desktop/render.c:35]: Believe this should be encompassed in the deco data struct

**Comment by WillPower3309** [comment]
> I'm currently not able to build this on my systems due to wlroots 0.16 not being released yet on Arch. I've just been building it with wlroots as a submodule. Is this something that we should consider doing as default?

For dev work I think it's a good idea, but for release we should stick with stable wlroots releases

**Comment by ErikReider** [comment]
Everything seems to be working as expected so I'll mark it as ready for review.

Currently the AUR git package will break if merged while the COPR package should be fine until you push a new release (I'll check tomorrow if/when wlroots 0.16 will be released). I'm not sure about the nix flake but I'm guessing that you have your eyes all over that issue

**Comment by ErikReider** [comment]
Seems like Fedora 38 (this spring) will ship with sway 1.8 and wlroots 0.16 so you'll have to untick the older fedora versions (<38). That should allow people still using older fedora versions to download and install the older compatible swayfx versions.

The Arch packages will update soon but I'm not really sure when...

**Comment by WillPower3309** [comment]
Thanks a ton! Once arch and nix are updated we can push this out

File: sway/desktop/render.c [hunk 2]
```diff
@@ -1299,6 +1299,41 @@ void output_render(struct sway_output *output, struct timespec *when,
...
+		float clear_color[] = {0.0f, 0.0f, 0.0f, 1.0f};
+		if (server.session_lock.lock == NULL) {
+			// abandoned lock -> red BG
+			clear_color[0] = 1.f;
+		}
+		int nrects;
+		pixman_box32_t *rects = pixman_region32_rectangles(damage, &nrects);
+		for (int i = 0; i < nrects; ++i) {
+			scissor_output(wlr_output, &rects[i]);
+			wlr_renderer_clear(renderer, clear_color);
```
Comment (fuegoio) [inline sway/desktop/render.c:1312]: Here I think it may be:
```suggestion
			fx_renderer_clear(clear_color);
```

What do you think?

**Comment by fuegoio** [comment]
Hi!

I tried on my own a few weeks ago to rebase the current work on the `master` branch of sway (which is now the stable release 1.8), however I ran in few issues that I tried to address. I looked at your changes but didn't see any addressing of those issues so I may have a few questions ahah:

* wlroots 0.16 deprecates the `wlr_egl_make_current` function as well as others EGL functions to make them private. This implies for me to change the headers to include private headers but I don't think it is a good way to go 🫤. Without  those functions either the build does not work or the software crashes. Did you have any issue regarding this?
* I had the same issue with the `wlr_matrix_projection`, but on my side I can't find the `matrix_projection` you are talking about, and if I try to build this branch I have no luck as well (build issue)

File: sway/desktop/render.c (hunk 2 above)
Comment (ErikReider) [reply sway/desktop/render.c:1312]: I'll check on this after the 🎉 weekend :)

File: sway/desktop/render.c [hunk 3]
```diff
@@ -1299,6 +1299,41 @@ void output_render(struct sway_output *output, struct timespec *when,
...
+		}
+
+		if (server.session_lock.lock != NULL) {
+			struct render_data data = {
+				.damage = damage,
+				.deco_data.alpha = 1.0f,
```
Comment (WillPower3309) [inline sway/desktop/render.c:1318]: We should probably init the `deco_data` with `get_undecorated_decoration_data()` (which also inits the alpha to 1.0f)

Comment (ErikReider) [reply sway/desktop/render.c:1318]: I'll fix this next year 🥁😂

**Comment by ErikReider** [comment]
> * wlroots 0.16 deprecates the `wlr_egl_make_current` function as well as others EGL functions to make them private. This implies for me to change the headers to include private headers but I don't think it is a good way to go 🫤. Without  those functions either the build does not work or the software crashes. Did you have any issue regarding this?
> 
> * I had the same issue with the `wlr_matrix_projection`, but on my side I can't find the `matrix_projection` you are talking about, and if I try to build this branch I have no luck as well (build issue)

Hmmm. The functions seems to be included in the wlroots source code but isn't included in the built header file. So compiling the correct wlroots version as a subproject worked when I tested it... Now when I've installed wlroots 0.16 as a system package, I'm getting the same issue :/

**Comment by ErikReider** [comment]
Fixed the usage of `wlr_egl_make_current` but I'm kinda stuck on `wlr_matrix_projection`. `matrix_projection` is private. Seems like Hyprland copies the same function into their project so I'll do the same

**Comment by ErikReider** [comment]
The latest wlroots and Sway are now available on Arch. What was the status of the Nix packages?

**Comment by WillPower3309** [comment]
Looks like it was merged, 1.8 is in nixpkgs unstable (although if we make a PR for nixpkgs with ours we would.be in unstable too)

**Review by WillPower3309 — APPROVED** [review]

**Pull request merged**